src/lib/publisher/term/change/index.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  PublisherTermChangeCancelParams,
+  PublisherTermChangeGetSubscriptionUpgradeStatusParams
+} from '../../../interfaces/api-params'
+import {
+  PublisherTermChangeCancelResponse,
+  PublisherTermChangeGetSubscriptionUpgradeStatusResponse
+} from '../../../interfaces/api-response'
+import { SubscriptionUpgradeStatus as ISubscriptionUpgradeStatus } from '../../../interfaces/subscription-upgrade-status';
+import { Piano } from '../../../piano';
+import { httpRequest } from '../../../utils/http-request';
+
+const ENDPOINT_PATH_PREFIX = '/publisher/term/change';
+
+export class Change {
+  constructor (private readonly piano: Piano) {}
+
+  /**
+   * Get upgrade status of subscription
+   *
+   * @see https://docs.piano.io/api/?endpoint=post~2F~2Fpublisher~2Fterm~2Fchange~2FgetSubscriptionUpgradeStatus
+   */
+
+  public async getSubscriptionUpgradeStatus(params: PublisherTermChangeGetSubscriptionUpgradeStatusParams): Promise<ISubscriptionUpgradeStatus | undefined> {
```

Comment:
`undefined` is the result returned from Piano's API when there is no upgrade associated with the subscription.